packages/web3-provider/src/subproviders/EthGasStationProvider.js

```diff
@@ -0,0 +1,59 @@
+const fetch = require('cross-fetch')
+const SubProvider = require('web3-provider-engine/subproviders/subprovider')
+
+const CACHE_TIME = 15000
+const GAS_STATION_URL = 'https://ethgasstation.info/json/ethgasAPI.json'
+const GAS_PRICE_KEY = 'safeLow'
+
+class EthGasStationProvider extends SubProvider {
+  constructor() {
+    super()
+
+    this.cache = {
+      data: null,
+      time: null
+    }
+  }
+
+  async fetchData() {
+    const res = await fetch(GAS_STATION_URL)
+    if (res.status !== 200) {
+      throw new Error(`Fetch returned code ${res.status}`)
+    }
+    const jason = await res.json()
+    if (typeof jason[GAS_PRICE_KEY] !== 'undefined') {
+      // values come from EGS as tenths of gwei
+      return String(jason[GAS_PRICE_KEY] * 1e8)
```

Comment:
These numbers won't ever be so big as to need BN?

Reply:
If gas price is ever at 9 Ether(max JS int) we have other issues.  Think this should throw in that case?  I really don't foresee a near future where that could be real.

Reply:
Ah yea no need to if it's 9 Eth